tests/cephfs/cephfs_mirroring/snapdiff_perf_improvements_across_releases.py

```diff
@@ -0,0 +1,783 @@
+import csv
+import os
+import random
+import string
+import time
+import traceback
+
+from ceph.ceph import CommandFailed
+from tests.cephfs.cephfs_mirroring.cephfs_mirroring_utils import CephfsMirroringUtils
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+log_base_dir = os.path.dirname(log.logger.handlers[0].baseFilename)
+log_dir = f"{log_base_dir}/SnapDiff_Results/"
+os.mkdir(log_dir)
+log.info(f"Log Dir : {log_dir}")
+
+csv_file = f"{log_dir}/snapshot_sync_info.csv"
+
+
+def initialize_csv_file():
+    try:
+        with open(csv_file, mode="x", newline="") as file:
+            writer = csv.writer(file)
+            writer.writerow(
+                [
+                    "Snapshot Type",
+                    "Snapshot Name",
+                    "Sync Duration",
+                    "Sync Timestamp",
+                    "Snaps Synced",
+                ]
+            )
+    except FileExistsError:
+        pass
+
+
+def log_snapshot_info(snapshot_type, snapshot_info):
+    with open(csv_file, mode="a", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(
+            [
+                snapshot_type,
+                snapshot_info["snapshot_name"],
+                snapshot_info["sync_duration"],
+                snapshot_info["sync_time_stamp"],
+                snapshot_info["snaps_synced"],
+            ]
+        )
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83595260 - Performance evaluation of snapdiff feature using CephFS Mirroring after upgrading to RHCS 8.0
+
+    Steps:
+        1. Prepare the Ceph clients and authenticate them on both clusters.
+        2. Set up CephFS volumes and MDS placement on both source and target clusters.
+        3. Enable CephFS mirroring modules and establish a peer connection between the two clusters.
+        4. Create subvolumes and mount them on kernel and FUSE clients.
+        5. Create directories for I/O operations.
+        6. Add paths for mirroring and create large files for data sync tests.
+        7. Take initial snapshots and validate the sync across the clusters.
+        8. Perform incremental snapshots after modifying files, and log the results.
+
+    Raises:
+        CommandFailed: If any command execution fails during the setup or snapshot process.
+
+    Logs:
+        - Snapshot information such as sync duration, timestamp, and synced snapshots are logged
+          to CSV files for both kernel and FUSE clients.
+
+    Returns:
+        int: 0 if the test completes successfully, or 1 if any critical step fails.
+    """
+    try:
+        config = kw.get("config")
+        ceph_cluster_dict = kw.get("ceph_cluster_dict")
+        fs_util_ceph1 = FsUtils(ceph_cluster_dict.get("ceph1"))
```

Comment:
Hi Hemanth,
Can you add this lines and pass test_data to the FsUtils
https://github.com/red-hat-storage/cephci/blob/738e64a6a7b6e07b6c23814a288bbc76670bedab/tests/cephfs/cephfs_mirroring/test_cephfs_mirror_auth_caps.py#L34

Reply:
Included @AmarnatReddy  - Please review it again.